core/src/main/java/org/jdeferred/impl/AbstractPromise.java

```diff
@@ -264,5 +292,8 @@ public void waitSafely(long timeout) throws InterruptedException {
 				}
 			}
 		}
+        if (thrownRuntimeException != null) {
+            throw thrownRuntimeException;
```

Comment:
The only place that throws a saved exception is inside `progress`. But what if an exception was saved during `triggerDone` and the client code did not register a `ProgressCallback`?